Develop (#2)
itk-dev/drupal-11-template: README.md

# Drupal 11 TEMPLATE

This is a Drupal project template for starting new projects. The template
includes default setup and a starting point for:

- PHP Code sniffer
- PHP Code styles
- Github actions
- Tasks
- Local settings
- Docker and overrides
- Readme
- Changelog

## Usage

### Build new project from template

Build a new project using this template

``` shell

> gh repo create itk-dev/{PROJECT-NAME} --template itk-dev/drupal-11-template --public --clone 

Reviewer: You get extra points for doing this with `gh`! 🎉

```

### Prune the project

> Prune the project to remove/override template specific files and setup the
> project, with docker config and settings.

Reviewer: ```suggestion
Prune the project to remove/override template specific files and set up the
project with docker config and settings:
```

``` shell
task template-prune PROJECTNAME={PROJECTNAME}
```

> 

Reviewer: ```suggestion

> [!IMPORTANT]  
> Replace `{PROJECTNAME}` with your actual project name.
```

This will:

- Remove TEMPLATE_THEME and cause coding-standards-check to fail until a new
theme is added.
- Add/override .env and .task.env files with files that uses {PROJECTNAME}
- Add/override a settings.local.php to web/sites/default folder
- Add/override a services.local.yml to web/sites/default folder
- Override CHANGELOG.md
- Override Taskfile.yml

### Finish

> - Replace {PROJECT-NAME} placeholder in this README.md and CHANGELOG.md

Reviewer: Why doesn't the `template-prune` task do this?

Author: Just a matter of time...

<!-- markdownlint-disable MD025 -->

> ### ---- After building a project delete this line and all above ----

Reviewer: The `template-prune` task could to this.

Author: Just a matter of time...

# Readme for {PROJECT-NAME}

> ## Build assets

Reviewer: This heading seems to be in the wrong place.

Author: No. This will be the h1 of the project template Readme in the newly created project.

> ### Site installation

Reviewer: This heading should be on the second level:

```suggestion
## Site installation
```

> Run the following commands to set up the site a new. This will start contaners

Reviewer: ```suggestion
Run the following commands to set up the site a new. This will start containers
```

and run composer install, add a settings.php file run and site-install.

``` shell name="site-up"
task build-site:new
```

If the site has existing config and a settings.php file build the site from that.

``` shell name="site-up"
task build-site:existing-conf
```

> When the installation is completed, that admin user is created and the password for logging in the outputted. If you
> forget the password, use drush uli command to get a one-time-login link (not the uri here only works if you are using
> trafik).

Reviewer: ```suggestion
When the installation is completed, that admin user is created and the password for logging in the outputted. If you
forget the password, use `drush user:login` command to get a one-time-login URL (note: the URI here only works if you are using
Traefik and [ITK-dev docker setup](https://github.com/itk-dev/devops_itkdev-docker)).
```

``` shell name="site-login"

> itkdev-docker-compose drush user:login

Reviewer: Consider adding a `drush` task for this.

I've played around with this in <https://github.com/rimi-itk/dpl-cms-develop/blob/develop/Taskfile.yml#L105-L108>. The site URL (well, the domain) is already set in `.env` and can be used in `Taskfile.yml` (cf. https://github.com/rimi-itk/dpl-cms-develop/blob/develop/Taskfile.yml#L4).

Author: Lots of tasks could be added. I agree this is one

```

### Access the site

If you are using out `itkdev-docker-compose` simple use the command below to åbne the site in you default browser.

``` shell name="site-open"
itkdev-docker-compose open
```

Alternatively you can find the port number that is mapped nginx container that server the site at `http://0.0.0.0:PORT`
by using this command:

``` shell
open "http://$(docker compose port nginx 8080)"

> ```

Reviewer: Consider adding a `open` (or `site:open`) task for this (and maybe a `site:url` helper task).

Until we start making real open source projects, we can assume that our https://github.com/itk-dev/devops_itkdev-docker setup is running, and use `COMPOSE_DOMAIN` from `.env` to get the site URL (cf. my comment on adding a `drush` task).

Author: Good idea, for future development

### Drupal config

This project uses Drupal's configuration import and export to handle configuration changes and uses the [config
ignore](https://www.drupal.org/project/config_ignore) module to protect some of the site settings form being overridden.
For local and production configuration settings that you do not want to export, please use `settings.local.php` to
override default configuration values.

Export config created from drupal:

```shell

> itkdev-docker-compose drush config:export

Reviewer: Consider using `task drush -- config:export` (if that task has been added).

Author: Also a missing task

```

Import config from config files:

``` shell
itkdev-docker-compose drush config:import
```

### Coding standards

``` shell name=coding-standards-composer

> docker compose exec phpfpm composer install
> docker compose exec phpfpm composer normalize

Reviewer: Consider adding a `compose` task (cf. https://github.com/rimi-itk/itk-fiware/blob/feature/pep-proxy/Taskfile.yml#L27-L29):

```suggestion
task compose -- exec phpfpm composer install
task compose -- exec phpfpm composer normalize
```
